cache/remotecache/import.go

```diff
@@ -97,6 +150,37 @@ func (ci *contentCacheImporter) Resolve(ctx context.Context, desc ocispecs.Descr
 	return solver.NewCacheManager(ctx, id, keysStorage, resultStorage), nil
 }
 
+// extends support for "new"-style image-manifest style remote cache manifests and determining downstream
+// handling based on inference of document structure (is this a new or old cache manifest type?)
+func inferManifestType(ctx context.Context, dt []byte) (ManifestType, error) {
```

Comment:
We already have a function called `DetectManifestMediaType`

Reply:
Heh, I feel very lucky it already exists :) Good learning for me to see a better quality implementation as well. I also like that it uses the containerd/oci mediatypes which seems more direct and extensible.